src/snapshot.c

```diff
@@ -883,3 +866,21 @@ void archiveSnapshot(RedisRaftCtx *rr)
              "%s.bak.%d", rr->config.rdb_filename, raft_get_nodeid(rr->raft));
     rename(rr->config.rdb_filename, bak_rdb_filename);
 }
+
+void SnapshotInit(RedisRaftCtx *rr)
+{
+    rr->outgoing_snapshot_file = (SnapshotFile){0};
+
+    /* Generate temp file name for incoming snapshots */
+    snprintf(rr->incoming_snapshot_file, sizeof(rr->incoming_snapshot_file),
+             "%s.tmp.recv", rr->config.rdb_filename);
+
+    /* Delete if there is a partial snapshot file from the previous run */
+    int ret = unlink(rr->incoming_snapshot_file);
+    if (ret != 0 && errno != ENOENT) {
+        PANIC("Unlink file: %s, error: %s \n", rr->incoming_snapshot_file,
+              strerror(errno));
+    }
+
+    resetSnapshotState(rr);
+}
```

Comment:
moved a few pieces here, this function is to be called on module load once.

Reply:
I wonder why - can you elaborate a little bit?

Reply:
https://github.com/RedisLabs/redisraft/pull/587/files#diff-d31a5eb8cc6de9b16ed710fd09c8a8516aa375c97f345fa98adf28ab4bf5b65eL2071

We initialize these metrics with `-1` there in `RedisRaftCtxInit()`. We also have `resetSnapshotState()`. So, I thought we may want a single init function. Metrics will be initialized/cleared by only `resetSnapshotState()`

Also, we have `initSnapshotTransferData()` function, so we can move that one here as well and call once on module load. My intention was to have a single place to init snapshot related variables.